File: python/code_examples/06-iteration/main.py
nums = [1, 2, 3, 4]

# List comprehension (Pythonic)
squares = [n * n for n in nums if n % 2 == 0]

# for-loop
total = 0
for n in nums:
    total += n


def main() -> None:
    print("squares(even):", squares, "total:", total)


if __name__ == "__main__":
    main()

File: python/code_examples/06-iteration/main.js
const nums = [1, 2, 3, 4];

const squares = nums.filter((n) => n % 2 === 0).map((n) => n * n);

let total = 0;
for (const n of nums) {
  total += n;
}

function main() {
  console.log("squares(even):", squares, "total:", total);
}

main();
